runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/engine/internal/ManagedHttpClientEngine.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.http.engine.internal
+
+import aws.smithy.kotlin.runtime.http.engine.HttpClientEngine
+import aws.smithy.kotlin.runtime.io.SdkManagedCloseable
+import aws.smithy.kotlin.runtime.util.InternalApi
+
+@InternalApi
+public class ManagedHttpClientEngine(
```

Comment:
**Question**: I think this can be `internal` or perhaps even `private` now. We only care about `SdkManaged` as a marker interface on the type, the concrete implementation doesn't need to be public anymore I don't think.